packages/expo/src/takeSnapshotAsync/Creator.web.ts

```diff
@@ -0,0 +1,289 @@
+  return await canvas.toDataURL('image/jpeg', quality);
+}
+
+export async function createBlobAsync(
+  element: Element,
+  { quality, ...options }: CaptureOptions
+): Promise<Blob> {
+  const canvas = await draw(element, options);
+  return await util.getBlobFromCanvasAsync(canvas, quality);
+}
+
+async function draw(
+  element: Element,
+  options: {
+    width?: number;
+    height?: number;
+    bgcolor?: string | CanvasGradient | CanvasPattern;
+    [key: string]: any;
+  }
+): Promise<HTMLCanvasElement> {
+  const fromSVG = await generateSVGAsync(element, options);
+  const nextV = await util.getImageElementFromURIAsync(fromSVG);
+  const image: any = await util.postponeMethodExecution(100)(nextV);
+
+  const canvas = newCanvas(element, options);
+  const context = canvas.getContext('2d');
+  if (context == null) {
+    throw new Error('Canvas context is not supported.');
+  }
+  context.drawImage(image, 0, 0);
+  return canvas;
+}
+
+function newCanvas(
+  element: Element,
+  options: { width?: number; height?: number; bgcolor?: string | CanvasGradient | CanvasPattern }
+) {
+  const canvas: HTMLCanvasElement = document.createElement('canvas');
+  canvas.width = options.width || util.getWidthForElement(element);
+  canvas.height = options.height || util.getHeightForElement(element);
+
+  if (options.bgcolor) {
+    const ctx = canvas.getContext('2d');
+    if (ctx) {
+      ctx.fillStyle = options.bgcolor;
+      ctx.fillRect(0, 0, canvas.width, canvas.height);
+    }
+  }
+
+  return canvas;
+}
+
+async function getDeepCopyForElement(
+  element: HTMLCanvasElement | Element
+): Promise<HTMLImageElement | Node> {
+  if (element instanceof HTMLCanvasElement) {
+    const dataURL = element.toDataURL();
+    return util.getImageElementFromURIAsync(dataURL);
+  }
+  return element.cloneNode(false);
+}
+
+async function cloneElement(
+  element: HTMLCanvasElement | Element
+): Promise<HTMLElement | undefined> {
+  const clonedNode = await getDeepCopyForElement(element);
+  const clone = await cloneChildren(element, clonedNode);
+  return await processClone(element, clone as any);
+}
+
+async function cloneChildren(
+  { childNodes },
+  clone: HTMLImageElement | Node
+): Promise<HTMLElement | Node> {
+  const children = util.makeIterable(childNodes);
+  if (children.length === 0) {
+    return clone;
+  }
+
+  for (const child of children) {
+    const childClone = await cloneElement(child);
+    if (childClone) {
+      await clone.appendChild(childClone);
+    }
+  }
+
+  return clone;
+}
+async function processClone(original: Element, clone: HTMLElement): Promise<HTMLElement | any> {
+  if (!(clone instanceof HTMLElement)) {
+    // TODO: Bacon: Avoid or throw error
+    return clone;
+  }
+
+  const source = window.getComputedStyle(original);
+  const target = clone.style;
+
+  if (source.cssText) {
+    target.cssText = source.cssText;
+  } else {
+    util.makeIterable(source).forEach(name => {
```

Comment:
Instead of makeIterable, the language can do that for you (supports array-like objects like NodeLists):

```
for (const name of source) {
  target.setProperty(...)
}
```